src/auth/user.zig

```diff
@@ -1,9 +1,28 @@
-//! This is a default User provided by Verse. This is almost certainly not what
-//! you want.
-user_ptr: ?*anyopaque,
+//! Auth.User is a thick user wrapper used within Auth.Provider and the other
+//! default authentication providers within Verse. Many of the fields here are
+//! provided here for convenience and aren't (currently) used within the
+//! Providers.
 
-/// deprecated do not use
-username: []const u8 = undefined,
+/// Reserved for callers. Never modified by any Verse Provider.
+user_ptr: ?*anyopaque = null,
+/// Reserved for callers.
+username: ?[]const u8 = null,
+
+// The following fields are used and modified by Verse Providers.
+
+/// The **currently* active session for the user. The session that was used to
+/// create the object. Often or likely included with the current user request.
+/// May be the same as session_next.
+/// See also: next_session
+session_current: ?[]const u8 = null,
```

Comment:
Should we create something like a SessionStore that is separate from the user and stores their session there. We could do a map like user id or username to the session. Doesn't have to be implemented now if you agree, I'm just thinking for in the future I know when the auth interface was first introduced we talked about allowing the user to bring their own user type.

Reply:
yeah, had a very similar thought to that, twice actually, once when writing the name, and wondering if we shouldn't have a session, and auto enroll every browser into a stateful connection; and then again later when I was calculating the cost of the constant hashing/validation steps.

Sessions are more complicated than tokens, so yes, but let's give it some thought